src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -175,6 +175,46 @@ public static IResourceBuilder<T> WithEnvironment<T>(this IResourceBuilder<T> bu
         return builder.WithAnnotation(new CommandLineArgsCallbackAnnotation(callback));
     }
 
+    /// <summary>
+    /// Adds a callback to be executed with a list of arguments to add to the container run command when a container resource is started.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="args">The arguments to be passed to the container run command when the container resource is started.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithContainerRunArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : IResourceWithArgs
```

Comment:
@danegsta I think we want these methods constrained on container resource and moved to container resource extensions.